### PR TITLE
Add typing to amazon provider EMR

### PR DESCRIPTION
**Pull request by Swalloow**
related: #9708, #9969
Adding type annotations to `amazon/emr`.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr_add_steps.py [hunk 1]
```diff
@@ -52,34 +53,36 @@ class EmrAddStepsOperator(BaseOperator):
...
-        job_flow_name=None,
-        cluster_states=None,
-        aws_conn_id='aws_default',
-        steps=None,
+        job_flow_id: Optional[str] = None,
+        job_flow_name: Optional[str] = None,
+        cluster_states: Optional[List[str]] = None,
+        aws_conn_id: str = 'aws_default',
+        steps: Optional[Union[List[dict], str]] = None,
+        *args,
```
Comment (turbaszek) [inline airflow/providers/amazon/aws/operators/emr_add_steps.py]:
```suggestion

```
As we cannot pass `args` right?

**Review by turbaszek — COMMENTED** [review]
Look good but we should remove `*args` from operators as we accept only key words arguments

File: airflow/providers/amazon/aws/operators/emr_add_steps.py [hunk 2]
```diff
@@ -52,34 +53,36 @@ class EmrAddStepsOperator(BaseOperator):
     @apply_defaults
     def __init__(
         self,
-        *,
```
Comment (feluelle) [inline airflow/providers/amazon/aws/operators/emr_add_steps.py:55]: We still need that `*` which enforces the user to use kwargs only.

File: airflow/providers/amazon/aws/operators/emr_add_steps.py (hunk 1 above)
Comment (feluelle) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: Yes, we shouldn't allow to pass `*args` to operators init. That's why we have the `*`.

**Comment by mik-laj** [comment]
We have conflicts. Can you do a rebase?

**Comment by Swalloow** [comment]
> We have conflicts. Can you do a rebase?

@mik-laj done

**Review by turbaszek — APPROVED** [review]

**Pull request merged**